### ISSUE TITLE
setFilesToUpload methos not working

### ISSUE THREAD

**Issue by niharika377** [DevExpress/testcafe#4571]
I am using testcafe for automating end to end scenario for a webportal . There is a scenario to upload file onto a "Choose File" button in the portal . I habve used setFilesToUpload method in order to do that , but the file is getting uploaded in the UI and testcase is showing as "Pass".

Below is the function I am calling in my testcafe 👍 
====================================

 async clickuploadBtn(t) {
       await t.setFilesToUpload(this.main),"/usr/local/html/tests/ui/test/e2e/helpers/resources/test_file.pdf")
        }

Also, I have defined "main" in the same class file like below:
============================
private main: Selector = Selector('div[id="mbUpload"]>div>div div~div>input');

Also,  Below is the html code for UI where I have to upload the file
========================================
<div id="mbUpload">
                           <div class="_uploadFileFormField">
   

   <input type="hidden" name="type_document_1_Group925-0_1" value="manitoba_consent_doc">

   <input type="hidden" name="country_document_1_Group925-0_1" value="">

      <div class="_uploadFileContainer">
      <div class="_existingFile" style="display: none">
            <label>
                     Scanned Copy of Manitoba Consent
                           <span class="dbLinkSpacer">|</span> <a href="#" onclick="clickRemoveFileUpload(this, '/screening/screening.ajax.php?op=removefileupload&amp;file=file_document_1_Group925-0_1'); return false;" class="smalltext">Remove</a>
               </label>
         <strong><input type="image" src="https://portal.talentwise.com/0f8e28ddea/images/paperclip.gif" style="vertical-align: -3px" alt=""></strong>
         </div>
      <div class="_uploadFile searchField searchFieldFullWidth " style="display: block">
         <label for="file_document_1_Group925-0_1" class="">Attach Scanned Copy of Manitoba Consent⁠*
</label>
         <input type="file" id="file_document_1_Group925-0_1" name="file_document_1_Group925-0_1" size="40" style="width: 200px" data-document-group="">
                  <span>&nbsp;(2MB)</span>
                           </div>
   </div>
==================================================
![Capture](https://user-images.githubusercontent.com/58839267/70774244-a4328000-1d9f-11ea-9df6-3a9773ec8e11.JPG)

 I am using tescafe version : 1.6.0
node js version : v8.16.0
npm version: 6.12.1

**Comment by miherlosev** [comment]
Hi @niharika377 

I am afraid this information is not sufficient to find the cause of the issue. Could you please provide a  small example so that we can reproduce it on our side?

**Comment by niharika377** [comment]
**There is a "Choose File" button in the UI  where I want to upload File of pdf format. Below is input tag for the choose file button.**

<input type="file" id="file_document_1_Group925-0_1" name="file_document_1_Group925-0_1" size="40" style="width: 200px" data-document-group="">
                  <span>&nbsp;(2MB)</span>

**But , the complete html snippet which leads to this input tag being visible is below:**

<div id="mbUpload">
<div class="_uploadFileFormField">
   

<input type="hidden" name="type_document_1_Group925-0_1" value="manitoba_consent_doc">

   <input type="hidden" name="country_document_1_Group925-0_1" value="">
<div class="_uploadFileContainer">
      <div class="_existingFile" style="display: none">
            <label>
                     Scanned Copy of Manitoba Consent
                           <span class="dbLinkSpacer">|</span> <a href="#" onclick="clickRemoveFileUpload(this, '/screening/screening.ajax.php?op=removefileupload&amp;file=file_document_1_Group925-0_1'); return false;" class="smalltext">Remove</a>
               </label>
         <strong><input type="image" src="https://portal.talentwise.com/0f8e28ddea/images/paperclip.gif" style="vertical-align: -3px" alt=""></strong>
         </div>
      <div class="_uploadFile searchField searchFieldFullWidth " style="display: block">
         <label for="file_document_1_Group925-0_1" class="">Attach Scanned Copy of Manitoba Consent⁠*
</label>
         <input type="file" id="file_document_1_Group925-0_1" name="file_document_1_Group925-0_1" size="40" style="width: 200px" data-document-group="">
                  <span>&nbsp;(2MB)</span>
                           </div>
   </div>



**Below is my testcase code from where I am calling a method 'clickuploadBtn()' which is part of another file**



import * as base from '../../base';

import CandidatePortalPage from "../../pages/candidatePortal/CandidatePortalPage";

fixture('Tests')// declare the fixture
    .page(base.getCustomerDashboardPage() + '/screening/login.php')
    .before(async ctx => { })
    .beforeEach(async ctx => {});


const candidatePortalPage = new CandidatePortalPage();
test('Canada Drivers Abstract Testcase', async t => {
   
   
    await candidatePortalPage.navigateToCandidatePage(t,"https:/someUrl.com");
    await candidatePortalPage.enterCandidateTempPassword(t,'Test');
    await candidatePortalPage.clickSignInBtn(t);
    await candidatePortalPage.clickLaunchTaskBtn(t);
    await candidatePortalPage.canadaDriverProvinceSelection(t,'MB','925','0','1');
    await candidatePortalPage.clickuploadBtn(t);

});



**Now , the clickuploadBtn () method is written in another file 👍** 

import {Selector} from "testcafe";
import { ClientFunction } from 'testcafe';
export default class CandidatePortalPage {

 private main: Selector = Selector('div[id="mbUpload"]>div>div div~div>input');

 async clickuploadBtn(t) {
	await t.setFilesToUpload(this.main), "/usr/local/html/tests/ui/test/e2e/helpers/resources/test_file.pdf")
 }     
} 





And , I have a pdf file in the location "/usr/local/html/tests/ui/test/e2e/helpers/resources/test_file.pdf".

When I am executing this testcase , it is showing as "Passed " in the console ,, but there is no file.pdf uploaded in the UI.

**Comment by miherlosev** [comment]
Could you please provide a complete example that I can run locally without additional set up?
